### PR TITLE
0.8

### PR DESCRIPTION
**Pull request by msuterski**
Closes #9.

Updates the plugin to work with drone 0.8 syntax.

A lot of the pieces are straight copy/paste from https://github.com/drone-plugins/drone-google-cloudstorage/pull/9 but simplified.

Removed the retry and backoff functionality from the plugin since the GCS client itself implement both now.

https://godoc.org/cloud.google.com/go/storage
https://cloud.google.com/storage/docs/exponential-backoff

The backoff and number of retries seems not to be configurable.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,40 +1,33 @@
...
   docker:
-    username: $$DOCKER_USER
-    password: $$DOCKER_PASS
-    email: $$DOCKER_EMAIL
-    repo: plugins/drone-gcs
+    image: plugins/docker
+    pull: true
+    secrets: [ docker_username, docker_password ]
+    repo: plugins/gcs
     tag: latest
```
Comment (msuterski) [inline .drone.yml]: I wasn't sure what the versioning should look like. Let me know and I can update to whatever works best.

Comment (tboerger) [reply .drone.yml]: You could look at https://github.com/drone-plugins/drone-google-cloudstorage/blob/7ca304836e70dbd1e09dd23b83a3886b0753d43b/.drone.yml (which comes from #11), this should be more or less the standard drone config for repos within this org.

Comment (msuterski) [reply .drone.yml]: Thanks for pointing me in the right direction. 

Can we get the #11 merged and I'll rebase this with updated master? I would prefer not to expand the scope of this PR. 

Also, if this is a new standard to be followed, can the https://github.com/msuterski/drone-plugin-starter be updated? It's very outdated.

Comment (tboerger) [reply .drone.yml]: > Also, if this is a new standard to be followed, can the https://github.com/msuterski/drone-plugin-starter be updated? It's very outdated.

Yeah, this should also get updated :)

**Comment by tboerger** [comment]
I have merged #11 now, please rebase your work.

**Comment by msuterski** [comment]
Updated the PR with latest changes in master.

Just FYI, the versioning on tagging (`build` vs `version`) in the `.drone.yml` file does not work or apply in this plugin. There isn't any `version` variable available (in master or this branch) to pass the version to the compiler. I looked at few other plugins and they also have broken build scripts in the same way. Other plugins don't have the `version` set in the build process at all, so this seems to be new.

**Comment by msuterski** [comment]
@tboerger do you need anything else to review this?

**Comment by msuterski** [comment]
@tonglil ^

File: DOCS.md [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
-* `cache_control` - Cache-Control header
-* `metadata` - an arbitrary dictionary with custom metadata applied to all objects
+* `source` - (Required) location of files to upload
+* `target` - (Required) destination to copy files to, including bucket name
+* `ignore` - (Optional) skip files matching this [pattern](https://golang.org/pkg/path/filepath/#Match), relative to `source`
+* `acl` - (Optional) a list of access rules applied to the uploaded files, in a form of `entity:role`
+* `gzip` - (Optional) files with the specified extensions will be gzipped and uploaded with "gzip" Content-Encoding header
+* `cache_control` - (Optional) Cache-Control header
+* `metadata` - (Optional) an arbitrary dictionary with custom metadata applied to all objects
+* `google_credentials` - (Required) is the GCP service account auth key
```
Comment (tonglil) [inline DOCS.md]: This can be `auth_key` as well.

File: plugin.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
+// Copyright 2015 Google Inc
```
Comment (tonglil) [inline plugin.go]: I don't think this is necessary here.

File: plugin.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+	if _, err := io.Copy(w, r); err != nil {
+		return err
+	}
+	return w.Close()
+}
+
+// gzipper returns a stream of file and a boolean indicating
+// whether the stream is gzip-compressed.
+//
+// The stream is compressed if vargs.Gzip contains file extension.
```
Comment (tonglil) [inline plugin.go]: I think `vargs` -> `p`

File: plugin.go [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+	if ext == "" {
+		return false
+	}
+	ext = ext[1:]
+	i := sort.SearchStrings(p.Gzip, ext)
+	return i < len(p.Gzip) && p.Gzip[i] == ext
+}
+
+// walkFiles creates a complete set of files to upload
+// by walking vargs.Source recursively.
```
Comment (tonglil) [inline plugin.go]: `vargs` -> `p`

File: DOCS.md (hunk 2 above)
Comment (msuterski) [reply DOCS.md]: Thanks. Missed that one. 

Maybe in the future it could be one of them?

Comment (tonglil) [reply DOCS.md]: What do you mean it could be?

Comment (tonglil) [reply DOCS.md]: Gotcha.
The best time to make that decision is now.
My preference is that all GCP plugins align on `GOOGLE_CREDENTIALS`.

File: plugin.go (hunk 3 above)
Comment (tonglil) [reply plugin.go]: I would just leave it in main.go.

**Comment by tonglil** [comment]
PTAL @tboerger :)

**Comment by tboerger** [comment]
I have merged that now, but the repo needs another push to the master branch because there had been an issue with the webhook configuration -.-

**Pull request merged**